helios-client/src/main/java/com/spotify/helios/common/descriptors/DeploymentGroupTasks.java

```diff
@@ -22,42 +22,57 @@
 package com.spotify.helios.common.descriptors;
 
 import com.google.common.base.Objects;
+import com.google.common.base.Optional;
+import com.google.common.collect.Sets;
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import java.util.Collections;
 import java.util.List;
+import java.util.Set;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class DeploymentGroupTasks extends Descriptor {
 
+  private static final Set<String> EMPTY_FAILED_TARGETS = Collections.emptySet();
+
   private final List<RolloutTask> rolloutTasks;
   private final int taskIndex;
   private final DeploymentGroup deploymentGroup;
+  private final Set<String> failedTargets;
+  private final int numTargets;
 
   private DeploymentGroupTasks(
       @JsonProperty("rolloutTasks") final List<RolloutTask> rolloutTasks,
       @JsonProperty("taskIndex") final int taskIndex,
-      @JsonProperty("deploymentGroup") final DeploymentGroup deploymentGroup) {
+      @JsonProperty("deploymentGroup") final DeploymentGroup deploymentGroup,
+      @JsonProperty("numTargets") final int numTargets,
```

Comment:
Note: It being 0 is fine -- it just means that any host failing will cause the DG to fail.